### PR TITLE
Added services and store to create and retrieve credentials. AB#10619

### PR DESCRIPTION
**Pull request by FuriousLlama**
# Fixes or Implements [AB#10619](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10619)

-   [x] Enhancement
-   [ ] Bug
-   [ ] Documentation

## Description

Added services and store for credential operations

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/src/store/modules/credential/actions.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        const logger: ILogger = container.get(SERVICE_IDENTIFIER.Logger);
+        const credentialService: ICredentialService =
+            container.get<ICredentialService>(
+                SERVICE_IDENTIFIER.CredentialService
+            );
+        return new Promise((resolve, reject) => {
+            if (context.getters.patientData.hdid !== undefined) {
+                logger.debug(
+                    `Credential Connection found stored, not querying!`
+                );
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/store/modules/credential/actions.ts:19]: I don't follow how this condition catches the case where the connection already exists.

File: Apps/WebClient/src/ClientApp/src/store/modules/credential/actions.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        });
+    },
+    retrieveCredentials(context, params: { hdid: string }): Promise<boolean> {
+        const logger: ILogger = container.get(SERVICE_IDENTIFIER.Logger);
+        const credentialService: ICredentialService =
+            container.get<ICredentialService>(
+                SERVICE_IDENTIFIER.CredentialService
+            );
+        return new Promise((resolve, reject) => {
+            if (context.getters.patientData.hdid !== undefined) {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/store/modules/credential/actions.ts:42]: Same comment as above.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/ClientApp/src/services/interfaces.ts [hunk 3]
```diff
@@ -12,6 +12,10 @@ import {
     ExternalConfiguration,
     OpenIdConnectConfiguration,
 } from "@/models/configData";
+import {
+    CredentialConnection,
+    WalletCredential,
```
[the remark this answers is not in the document]
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/services/interfaces.ts:17]: I don't think we have this set this in stone anywhere, for now I think they can stay as is and we can rename it later.

File: Apps/WebClient/src/ClientApp/src/services/restCredentialService.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    private readonly CREDENTIAL_BASE_URI: string = "/v1/api/Credential";
+    private http!: IHttpDelegate;
+    private isEnabled = false;
+
+    public initialize(
+        config: ExternalConfiguration,
+        http: IHttpDelegate
+    ): void {
+        this.http = http;
+        this.isEnabled = config.webClient.modules["Credential"];
```
[the remark this answers is not in the document]
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/services/restCredentialService.ts:32]: I think singular is better, same as the other modules.

File: Apps/WebClient/src/ClientApp/src/models/credentialConnection.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    credentialId: string;
+    WalletConnectionId: string;
+    IssuerCredentialId: string;
+    sourcetype: string;
+    sourceId: string;
+    AddedDate: StringISODateTime;
+    RevokedDate?: StringISODateTime;
+    status: CredentialState;
+    version: string;
+}
```
[the remark this answers is not in the document]
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/models/credentialConnection.ts:30]: sure, updated them

File: Apps/WebClient/src/ClientApp/src/store/modules/credential/actions.ts (hunk 2 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/store/modules/credential/actions.ts:42]: Yeah, same as the previous one

File: Apps/WebClient/src/ClientApp/src/services/restCredentialService.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/src/services/restCredentialService.ts:32]: Okay, makes sense.

File: Apps/WebClient/src/ClientApp/src/store/modules/credential/actions.ts (hunk 1 above)
Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/src/store/modules/credential/actions.ts]: Right, I meant the check on `patientData.hdid`.  But I see you've addressed that in the latest commit.

File: Apps/WebClient/src/ClientApp/src/services/interfaces.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/src/services/interfaces.ts:17]: Okay.  Will re-evaluate when the PR to merge the feature branch into dev is created.

File: Apps/WebClient/src/ClientApp/src/models/credentialConnection.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import { StringISODateTime } from "./dateWrapper";
+
+export interface CredentialConnection {
+    walletConnectionId: string;
+    hdid: string;
+    issuerConnectionid: string;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/models/credentialConnection.ts:6]: `issuerConnectionid` should be `issuerConnectionId`.

File: Apps/WebClient/src/ClientApp/src/models/credentialConnection.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    Pending,
+    Connected,
+    Disconnected,
+}
+
+export interface WalletCredential {
+    credentialId: string;
+    walletConnectionId: string;
+    issuerCredentialId: string;
+    sourcetype: string;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/models/credentialConnection.ts:24]: `sourcetype` should be `sourceType`.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/ClientApp/src/models/credentialConnection.ts (hunk 7 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/models/credentialConnection.ts:24]: The names are based on the discussion we have on https://qslvic.visualstudio.com/Health%20Gateway/_workitems/edit/10605.

**Review by MikeLyttle — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/models/credentialConnection.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
+import { StringISODateTime } from "./dateWrapper";
+
+export interface CredentialConnection {
+    walletConnectionId: string;
+    hdid: string;
```
Comment (sslaws) [inline Apps/WebClient/src/ClientApp/src/models/credentialConnection.ts:5]: We may have small tweaks to this but good for now.

**Review by sslaws — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/models/credentialConnection.ts (hunk 8 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/models/credentialConnection.ts:5]: Yeah, is a good enough approximation for now

**Pull request merged**